### PR TITLE
Remove functionbeat

### PR DESCRIPTION
**Pull request by AndersonQ**
## Proposed commit message

remove functionbeat

Starting at 9.0.0 functionbeat will not be released anymore. So this changes remove the functionbeat code.

## Checklist

- [x] My code follows the style guidelines of this project
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- [ ] I have made corresponding changes to the documentation
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Disruptive User Impact

- Functionbeat is no longer available

### Dependencies:
- [x] docs repo
  -  https://github.com/elastic/docs/pull/2895/
- [x] kibana repo
  - https://github.com/elastic/kibana/issues/193030
  - https://github.com/elastic/kibana/pull/198902
    - https://github.com/elastic/kibana/pull/199158
    - https://github.com/elastic/kibana/pull/199161
  - https://github.com/elastic/kibana/pull/199301
    - https://github.com/elastic/kibana/pull/200056
    - https://github.com/elastic/kibana/pull/200057
  - [x] remove telemetry/index patterns (or not?) -> [no](https://github.com/elastic/beats/issues/40745#issuecomment-2518829105)
- [x] Remove from DRA automation 
  - [x] https://github.com/elastic/infra/pull/41708

## How to test this PR locally

WIP

## Related issues

- Closes https://github.com/elastic/beats/issues/40745

### CODE REVIEW

File: dev-tools/cmd/update_go/update_go_version.go
Comment (AndersonQ) [inline dev-tools/cmd/update_go/update_go_version.go:1]: The lint issues aren't related to this PR, thus I'm not planning to fix them.

**Comment by jlind23** [comment]
Linter failures are not related to this PR.

**Comment by jlind23** [comment]
@cmacknz functionbeat has been deprecated for quite some time already, does it still deserve a changelog entry?

**Comment by cmacknz** [comment]
Functionbeat is still available for download appearing on https://www.elastic.co/downloads/beats and https://www.elastic.co/downloads/beats/functionbeat. This means that:

1. This deserves a changelog entry
2. Those pages on elastic.co need to remove functionbeat
3. There are likely parts of the unified release that expect functionbeat and those need to be removed before we merge this or we'll break the unified release eventually when the previous DRA artifacts expire.

**Comment by jlind23** [comment]
> 1. This deserves a changelog entry

I will create one right away.

> 2. Those pages on elastic.co need to remove functionbeat

This pages should only be removed for 9.0 and all versions above. @nimarezainia to whom should I reach out to for this?

> 3. There are likely parts of the unified release that expect functionbeat and those need to be removed before we merge this or we'll break the unified release eventually when the previous DRA artifacts expire.

This has already been taken care of by @AndersonQ

**Comment by cmacknz** [comment]
> There are likely parts of the unified release that expect functionbeat and those need to be removed before we merge this or we'll break the unified release eventually when the previous DRA artifacts expire.
> This has already been taken care of by @AndersonQ

I still see functionbeat in the beats artifact summary in the unified release though https://snapshots.elastic.co/9.0.0-f4176a12/summary-9.0.0-SNAPSHOT.html#beats

**Review by belimawr — APPROVED** [review]
Given that's mostly removing code and I cannot find any reference to `functionbeat` any more (aside from changelog entries): LGTM.

**Comment by jlind23** [comment]
> I still see functionbeat in the beats artifact summary in the unified release though https://snapshots.elastic.co/9.0.0-f4176a12/summary-9.0.0-SNAPSHOT.html#beats

@jmlrt you merged the release manager PR 20 hours ago so this is supposed to be fixed soon right?

**Comment by jmlrt** [comment]
> > I still see functionbeat in the beats artifact summary in the unified release though https://snapshots.elastic.co/9.0.0-f4176a12/summary-9.0.0-SNAPSHOT.html#beats
> 
> @jmlrt you merged the release manager PR 20 hours ago so this is supposed to be fixed soon right? Or do we need to manually remove the one that were already pushed?
> 
> cc @alsayasneh

@jlind23 https://snapshots.elastic.co/9.0.0-f4176a12/summary-9.0.0-SNAPSHOT.html#beats is from a build dating from December 3rd. https://github.com/elastic/unified-release/pull/1558 was merged 10 hours ago and the latest build doesn't include packetbeat anymore: https://snapshots.elastic.co/9.0.0-8585ff93/summary-9.0.0-SNAPSHOT.html

**Comment by jlind23** [comment]
> latest build doesn't include packetbeat

@jmlrt You meant functionbeat right? 😄 Do we need to clean up old snapshot or they'll be cleaned up once they reach their expiration delay?

@cmacknz according to @jmlrt comment I think we are good to proceed here right?

**Comment by jmlrt** [comment]
> @jmlrt You meant functionbeat right? 😄 Do we need to clean up old snapshot or they'll be cleaned up once they reach their expiration delay?

Indeed it's functionbeat 😅
Yes old snapshot will be cleaned following the bucket lifecycle policy (I think it's 1 month for snapshot)

**Review by rowlandgeoff — APPROVED** [review]
The Buildkite pipeline and GH workflow changes look ok. The packaging pipeline can't be tested in a PR though, so please watch that carefully after this merges.
https://buildkite.com/elastic/beats-packaging-pipeline

**Comment by jlind23** [comment]
@rowlandgeoff thanks for the heads up, packaging pipeline worked like a charm.

**Comment by dliappis** [comment]
After a discussion with @jlind23 we should remove functionbeat also in all active-8 branches.

**Pull request merged**
